tests/rtt_rosparam_tests/test/param_tests.cpp

```diff
@@ -606,6 +642,49 @@ TEST_F(ParamTest, SetValueOnParameterServer)
   compareValues(props, params, "set[Type]ComponentPrivate()", /* only_ros_types = */ true);
 }
 
+TEST_F(ParamTest, AddRosParamDataSources)
+{
+  // initialize properties to some values
+  props.initialize();
+
+  // Read in parameters through data source
+  ros::param::set("bool_parameter", props.bool_);
+  EXPECT_EQ(rosparam->addRosParamPropertyBool("bool_parameter").value(), props.bool_);
+  ros::param::set("double_parameter", props.double_);
+  EXPECT_EQ(rosparam->addRosParamPropertyDouble("double_parameter").value(), props.double_);
+  ros::param::set("float_parameter", props.float_);
+  EXPECT_EQ(rosparam->addRosParamPropertyFloat("float_parameter").value(), props.float_);
+  ros::param::set("int_parameter", props.int_);
+  EXPECT_EQ(rosparam->addRosParamPropertyInt("int_parameter").value(), props.int_);
+  ros::param::set("string_parameter", props.string_);
+  EXPECT_EQ(rosparam->addRosParamPropertyString("string_parameter").value(), props.string_);
+
+  // change the props values to an alternative configuration
+  props.initialize_alternative();
+
+  // Write out paramters through data source
+  bool in_bool;
+  double in_double;
+  float in_float;
+  int in_int;
+  std::string in_string;
+  dynamic_cast<RTT::Property<bool>*>(tc->getProperty("bool_parameter"))->set(props.bool_);
```

Comment:
nit: There is a special member function [RTT::PropertyBag::getPropertyType\<T>(name)](https://orocos-toolchain.github.io/rtt/toolchain-2.9/api/html/classRTT_1_1PropertyBag.html#a5d2825978bc3fee037bf39abc4f4648f) that includes the cast, but it is not exposed in `RTT::TaskContext` directly:

```suggestion
  tc->properties()->getPropertyType<bool>("bool_parameter")->set(props.bool_);
```
Whatever you prefer.